### ISSUE TITLE
Could trainer.validate don't enter _on_evaluation_model_train() ? 

### ISSUE THREAD

**Issue by rnwang04** [Lightning-AI/lightning#13775]
## 🚀 Feature

a trainer.validate() func which don't enter self._on_evaluation_model_train().

### Motivation

Here I have a trainer and a corresponding traced model and I don't want the model.train(model) in the end of trainer.validate().
How could I disasble changing to train mode again in the EvaluationLoop? 
```
# enable train mode again
self._on_evaluation_model_train()
```
Is there any convenient way to achieve this goal?

My PL version : 1.6.4

cc @akihironitta

**Comment by akihironitta** [comment]
Hi @rnwang04, you can achieve it by manually calling `.train()` before running any code you want to run with the train mode.
